### PR TITLE
4780 switch off private mode

### PR DESCRIPTION
**Pull request by Baron-Severin**
### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
Doesn't change the UI
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- [ ] **Milestone**: Make sure issues finished by this pull request are added to the [milestone](https://github.com/mozilla-mobile/fenix/milestones) of the version currently in development.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/FenixApplication.kt [hunk 1]
```diff
@@ -126,6 +130,17 @@ open class FenixApplication : Application() {
...
 
+    /**
+     * Clears private mode. This is done in order to avoid leaking the fact that
+     * private mode was in use during the previous session.
+     */
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    fun maybeClearPrivateMode(settings: Settings = settings()) {
+        // This needs to be called before the theme is set. No BrowsingModeManager is available
+        // at this point, which is why this is set directly
+        if (!settings.alwaysOpenInPrivateMode) settings.usePrivateMode = false
```
[the remark this answers is not in the document]
Comment (sblatz) [reply app/src/main/java/org/mozilla/fenix/FenixApplication.kt]: PR opened: https://github.com/mozilla-mobile/fenix/pull/5701/files

**Comment by Baron-Severin** [comment]
> This doesn't seem to be working for me:

To clarify, there are two flows here.

### Flow 1 (working as expected) 
STR:
- Open FF
- Set private mode
- Kill FF
- Reopen

Expected:
- FF opens in normal mode

Actual:
- FF opens in normal mode

### Flow 2 (failing) 
STR:
- Open FF
- Set private mode
- **Navigate to any website**
- Kill FF
- Reopen

Expected:
- No tabs are open
- FF opens in normal mode

Actual:
- No tabs are open
- **FF opens in private mode**


I'm working on a fix

**Comment by Baron-Severin** [comment]
It looks like the application is not being destroyed if the app is force killed while a private tab is opened.  Checked by running `adb shell ps | grep org.mozilla.fenix` after force killing with and without private tabs open.

Seems to be related to the private tab notification.  I commented out this notification, and the app was killed as expected.

**Comment by Baron-Severin** [comment]
@sblatz: @ boek suggested storing `usePrivateMode` in memory as an alternative to this approach.  I think that's a really great suggestion, but until we work out why the application is not being destroyed when the notification service is, it won't solve that edge case.  I haven't worked out a approach than this in the meantime, and I don't think it's worth spending more time on it.

**Pull request merged**